src/main/java/core/basesyntax/AbstractFigure.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class AbstractFigure implements Figure {
+    protected String color;
+
+    public AbstractFigure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
```

Comment:
It seems you've used an abstract class to set behavior for classes, which is not recommended in this task. Consider using an interface to define the behavior and use classes to implement this behavior. Abstract classes should be used when there are some common methods with default implementations that subclasses can inherit and possibly override, not just to define methods to be implemented.